iDiff/utils/image_prep_utils.go

```diff
@@ -182,10 +186,90 @@ func (p IDPrepper) ImageToFS() (string, error) {
 	return getImageFromTar(tarPath)
 }
 
+func (p IDPrepper) getConfig() (ConfigSchema, error) {
+	// check client compatibility with Docker API
+	valid, err := ValidDockerVersion()
+	if err != nil {
+		return ConfigSchema{}, err
+	}
+	var containerConfig container.Config
+	if !valid {
+		glog.Info("Docker version incompatible with api, shelling out to local Docker client.")
+		containerConfig, err = getImageConfigCmd(p.Source)
+	} else {
+		containerConfig, err = getImageConfig(p.Source)
+	}
+	if err != nil {
+		return ConfigSchema{}, err
+	}
+
+	config := ConfigObject{
+		Env: containerConfig.Env,
+	}
+	history := p.getHistory()
+	return ConfigSchema{
+		Config:  config,
+		History: history,
+	}, nil
+}
+
+func (p IDPrepper) getHistory() []ImageHistoryItem {
+	history, err := getImageHistory(p.Source)
+	if err != nil {
+		glog.Error("Could not obtain image history for %s: %s", p.Source, err)
+	}
+	historyItems := []ImageHistoryItem{}
+	for _, item := range history {
+		historyItems = append(historyItems, ImageHistoryItem{CreatedBy: item.CreatedBy})
+	}
+	return historyItems
+}
+
 type TarPrepper struct {
 	ImagePrepper
 }
 
-func (p TarPrepper) ImageToFS() (string, error) {
+func (p TarPrepper) getFileSystem() (string, error) {
 	return getImageFromTar(p.Source)
 }
+
+func (p TarPrepper) getConfig() (ConfigSchema, error) {
+	tmpDir := strings.TrimSuffix(p.Source, filepath.Ext(p.Source))
+	defer os.Remove(tmpDir)
+	err := UnTar(p.Source, tmpDir)
+	if err != nil {
+		return ConfigSchema{}, err
+	}
+	contents, err := ioutil.ReadDir(tmpDir)
+	if err != nil {
+		glog.Errorf("Could not read image tar contents: %s", err)
+		return ConfigSchema{}, errors.New("Could not obtain image config")
+	}
+
+	var config ConfigSchema
+	configList := []string{}
+	for _, item := range contents {
+		if filepath.Ext(item.Name()) == ".json" && item.Name() != "manifest.json" {
+			if len(configList) != 0 {
+				// Another <image>.json file has already been processed and the image config obtained is uncertain.
+				glog.Error("Multiple possible config sources detected for image at " + p.Source + ", some diff results may be incorrect.")
```

Comment:
I think we may have discussed this before, but this would be the case that the user saves an image specified by a repo without using a tag.  In this case, all the images in the repo are saved into a single tar, so there are config files for each one in the tarball.  I think Abby and I had previously discussed just getting the first one (because I don't think we can get all of them without running the risk of getting conflicting information between them) and warning the user that there were multiple possibilities so the information we're giving them may not be completely accurate.  Should we just not support tars that contain multiple images in one?